code/Pagetypes/FreshdeskPage.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+
+class FreshdeskPage extends Page
+{
+}
+
+class FreshdeskPage_Controller extends Page_Controller
+{
+    /**
+     * Priority mapping for Freshdesk tickets.
+     * @var array
+     */
+    private static $freshdeskPriority = [
+        0 => 'any',
+        1 => 'low',
+        2 => 'medium',
+        3 => 'high',
+        4 => 'urgent',
+    ];
+
+    private static $dependencies = [
+        'freshdesk' => '%$FreshdeskAPI',
+    ];
+
+    /**
+     * @var FreshdeskAPI
+     */
+    public $freshdesk;
+
+    /**
+     * @var array
+     */
+    private static $_tickets = [];
+
+    /**
+    * Returns Freshdesk tickets for a user
+    *
+    * @return PaginatedList
+    */
+    public function getTickets()
+    {
+        $currentMember = \Member::currentUser();
+        if (!$currentMember || !$currentMember->exists()) {
+            return \Security::permissionFailure();
+        }
+
+        if (!self::$_tickets) {
+            self::$_tickets = $this->freshdesk->getUserTickets($currentMember);
+        }
+
+        $tickets = self::$_tickets;
+
+        if (empty($tickets) || count($tickets) == 0) {
+            return false;
+        }
+
+        // Get all open by default
+        $filter = ['status' => 2];
+        if ($this->request->getVar('status')) {
+            $filter['status'] = (int) $this->request->getVar('status');
+        }
+        if ($this->request->getVar('priority')) {
+            $filter['priority'] = (int) $this->request->getVar('priority');
+        }
+
+        $tickets = $this->filterTickets($tickets, $filter);
```

Comment:
I still feel like you should try using `ArrayData` for filtering. Pack the data into that object, and then try using `filter` / `exclude`, instead of implementing your own routine (my hope is `filterTickets` can go away).

Reply:
That's what I'm doing, I kept `filterTickets` as a seperate method so I could write a test for it (:
https://github.com/jakedaleweb/silverstripe-freshdesk/blob/4a4331da7f1608a1309c92025af37f6dd4c789ef/code/Pagetypes/FreshdeskPage.php#L91

Reply:
Ah! Missed that :-)